homeassistant/components/panasonic_viera/__init__.py

```diff
@@ -203,47 +199,51 @@ async def async_send_key(self, key: Keys | str) -> None:
         except (AttributeError, TypeError):
             key = getattr(key, "value", key)
 
+        assert self._control is not None
         await self._handle_errors(self._control.send_key, key)
 
-    async def async_turn_on(self, context):
+    async def async_turn_on(self, context: Context | None) -> None:
         """Turn on the TV."""
         if self._on_action is not None:
             await self._on_action.async_run(context=context)
             await self.async_update()
-        elif self.state != STATE_ON:
+        elif self.state != MediaPlayerState.ON:
             await self.async_send_key(Keys.POWER)
             await self.async_update()
 
-    async def async_turn_off(self):
+    async def async_turn_off(self) -> None:
         """Turn off the TV."""
-        if self.state != STATE_OFF:
+        if self.state is not MediaPlayerState.OFF:
```

Comment:
We discussed that somewhere already which is better: an equality or identity check. Just don't remember which one was better 😅

Tbh it probably doesn't matter, although we should at least be consistent within a file.